rust/lancedb/src/index.rs

```diff
@@ -37,137 +36,276 @@ pub enum IndexParams {
     },
 }
 
-/// Builder for Index Parameters.
+/// Builder for creating an IVF PQ index.
+///
+/// See [VectorIndexBuilder::ivf_pq] for more details.
+pub struct IvfPqIndexBuilder {
+    parent: Arc<dyn TableInternal>,
+    pub(crate) common: IndexBuilder,
+
+    pub(crate) distance_type: DistanceType,
+    pub(crate) num_partitions: Option<u32>,
+    pub(crate) num_sub_vectors: Option<u32>,
+    pub(crate) sample_rate: u32,
+    pub(crate) max_iterations: u32,
+}
 
+/// Builder for creating some kind of index.
+///
+/// The methods on this builder are used to specify the type of index to create and return
+/// builders specific to that index type.
 pub struct IndexBuilder {
     parent: Arc<dyn TableInternal>,
-    pub(crate) columns: Vec<String>,
-    // General parameters
-    /// Index name.
-    pub(crate) name: Option<String>,
-    /// Replace the existing index.
+    pub(crate) columns: Option<Vec<String>>,
     pub(crate) replace: bool,
+}
+
+/// Builder for creating some kind of vector index.
+///
+/// See [IndexBuilder::vector] for more details.
+pub struct VectorIndexBuilder {
```

Comment:
Why are there separate vector vs scalar indices? Given that there are specific builders for BTree and IVF_PQ?

Reply:
These vector/scalar builders have gone away